### PR TITLE
ESSIFI-170 Add class-validator to pex-ms

### PR DESCRIPTION
**Pull request by zoemaas**
- Added class-validator to pex-ms;
   - Fixed bugs related to the entity id;
   - Using POST instead of PUT to update the status of the PE;

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,9 @@
   "dependencies": {
     "@sphereon/pex": "^0.6.3-unstable.1",
```
Comment (nklomp) [inline package.json]: 0.7.0 is out, which should not really impact the integration, but contains several fixes

File: src/entity/presentation/verifiableCredentialEntity.ts [hunk 2]
```diff
@@ -1,36 +1,60 @@
...
\ No newline at end of file
+  @Column()
+  @IsDefined({ message: 'VerifiableCredential.@context must be provided' })
+  @IsArray({ message: 'VerifiableCredential.@context must be an array' })
+  '@context': string[];
+
+  @Column()
+  @IsNotEmpty({ message: 'VerifiableCredential.id is invalid' })
+  //@ts-ignore
+  id: string;
```
Comment (nklomp) [inline src/entity/presentation/verifiableCredentialEntity.ts:17]: PK

File: src/entity/presentation/presentationWrapperEntity.ts [hunk 3]
```diff
@@ -1,20 +1,54 @@
...
+
+  @Column()
+  @IsDefined({ message: 'PresentationWrapper.thread must be provided' })
+  // @ts-ignore
+  thread: ThreadEntity;
+
+  @Column()
+  @IsNotEmpty({ message: 'PresentationDefinitionWrapper.id is invalid' })
+  // @ts-ignore
+  id: string;
```
Comment (nklomp) [inline src/entity/presentation/presentationWrapperEntity.ts:37]: PK

File: src/entity/presentation/presentationSubmissionEntity.ts [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
-  @Column((type) => DescriptorEntity)
+
+  @Column(() => DescriptorEntity)
+  @IsDefined({ message: 'PresentationSubmission.descriptor_map must be provided' })
+  @IsArray({ message: 'PresentationSubmission.descriptor_map must be an array' })
   descriptor_map: DescriptorEntity[];
+
   @Column()
+  @IsNotEmpty({ message: 'PresentationSubmission.id is invalid' })
   id: string;
```
Comment (nklomp) [inline src/entity/presentation/presentationSubmissionEntity.ts:19]: PK

File: src/entity/presentation/presentationEntity.ts [hunk 5]
```diff
@@ -1,22 +1,34 @@
...
-    verifiableCredential: VerifiableCredentialEntity[];
-    @Column(() => PresentationSubmissionEntity)
-    presentation_submission?: PresentationSubmissionEntity;
-    @Column()
-    holder?: string;
-    @Column()
-    proof?: IProof | IProof[];
-}
\ No newline at end of file
+  @Column()
```
Comment (nklomp) [inline src/entity/presentation/presentationEntity.ts:9]: Missing an autogenerated id that is PK

File: src/entity/presentationDefinition/presentationDefinitionWrapperEntity.ts [hunk 6]
```diff
@@ -14,14 +16,17 @@ export class PresentationDefinitionWrapperEntity {
...
   @Column()
+  @IsDefined({ message: 'PresentationDefinitionWrapper.presentation_definition must be provided' })
   // @ts-ignore
-  presentation_definition: IPresentationDefinition
+  presentation_definition: IPresentationDefinition;
 
   @Column()
+  @IsNotEmpty({ message: 'PresentationDefinitionWrapper.id is invalid' })
   // @ts-ignore
   id: string;
```
Comment (nklomp) [inline src/entity/presentationDefinition/presentationDefinitionWrapperEntity.ts:26]: PK

**Review by nklomp — CHANGES_REQUESTED** [review]
I probably should have more remarks, but given the deadline ok for the rest.
Missing some PKs and would update to pex 0.7.0

File: package.json (hunk 1 above)
Comment (zoemaas) [reply package.json]: I will update it.

File: src/entity/presentation/presentationWrapperEntity.ts (hunk 3 above)
Comment (zoemaas) [reply src/entity/presentation/presentationWrapperEntity.ts:37]: That one needs to match the presentation id afaik, so I will leave it as it is for the same reasons as above.

File: src/entity/presentation/presentationSubmissionEntity.ts (hunk 4 above)
Comment (zoemaas) [reply src/entity/presentation/presentationSubmissionEntity.ts:19]: Embedded document

File: src/entity/presentation/presentationEntity.ts (hunk 5 above)
Comment (zoemaas) [reply src/entity/presentation/presentationEntity.ts:9]: Embedded document

File: src/entity/presentationDefinition/presentationDefinitionWrapperEntity.ts (hunk 6 above)
Comment (zoemaas) [reply src/entity/presentationDefinition/presentationDefinitionWrapperEntity.ts:26]: That one needs to match the presentation id afaik, so I will leave it as it is

Comment (nklomp) [reply src/entity/presentationDefinition/presentationDefinitionWrapperEntity.ts:26]: Primary keys do not have to be autogenerated, they need to be unique (and in most cases are autogenerated).

**Review by nklomp — APPROVED** [review]
According to offline discussion, lets continue

**Pull request merged**